runtime/vm/vmbootlib.c

```diff
@@ -917,3 +925,58 @@ invoke31BitJNI_OnXLoad(J9JavaVM *vm, void *handle, jboolean isOnLoad, void *rese
 	return result;
 }
 #endif /* defined(J9VM_ZOS_3164_INTEROPERABILITY) && (JAVA_SPEC_VERSION >= 17) */
+
+#if defined(LINUX)
+// void seeIfIAmMissingAnything(int ret) {
+	// madvise() could fail with EINVAL if MADV_PAGEOUT is not supported by the kernel
+	// if (TR::Options::getCmdLineOptions()->getVerboseOption(TR_VerbosePerformance))
+		// TR_VerboseLog::writeLineLocked(TR_Vlog_PERF, "WARNING: Failed to use madvise to disclaim memory for data cache");
+	// printf("Failed to use madvise to disclaim memory for data cache, ret=%d\n", ret);
+	// if (ret == EINVAL)
+		// _disclaimEnabled = false; // Don't try to disclaim again, since support seems to be missing
+		// if (TR::Options::getCmdLineOptions()->getVerboseOption(TR_VerbosePerformance))
+			// TR_VerboseLog::writeLineLocked(TR_Vlog_PERF, "WARNING: Disabling data cache disclaiming from now on");
+	// }
+	// if ((9943 == ret) || (22 == ret)) {
+		// printf("Don't try to disclaim again, since support seems to be missing\n");
+	// }
+// }
+
+/**
+ * Note this funciton is specific to linux
+ *
+ * Disclaim all class loader memory segments.
+ *
+ * Walks all class loaders (via allClassLoadersStartDo, etc.) and calls madvise(segment->baseAddress, segment->size, MADV_PAGEOUT)
+ * on all the J9MemorySegments (classloader->classSegments)
+ *
+ * There is a chance that MADV_PAGEOUT is not supported on the OS level, so we need to the error code to see if it was successful
+ *
+ * @return 0 if successful, error code otherwise
+ */
+int
+disclaimAllClassMemory(J9VMThread *currentThread)
```

Comment:
actually, I think this function should be in jvmfree.c

Reply:
check if `J9_EXTENDED_RUNTIME2_DISCLAIM_RAM_CLASSES` is set, if not just return a failure. You can pick an error code that madvise doesn't return so you can tell it appart.